scopes/preview/preview/preview.main.runtime.tsx

```diff
@@ -138,10 +138,9 @@ export class PreviewMain {
 
       const map = await previewDef.getModuleMap(components);
       const environment = context.envRuntime.env;
-
-      const compilerInstance = hasCompiler(environment) && environment.getCompiler();
+      const compilerInstance = environment.getCompiler?.();
       const withPaths = map.map<string[]>((files, component) => {
-        const modulePath = this.pkg.getModulePath(component);
+        const modulePath = compilerInstance?.getPreviewComponentPath?.(component) || this.pkg.getModulePath(component);
```

Comment:
what did you do later with the 
`getDistPathBySrcPath` (few lines below)?
because I guess on your compiler you are going to return the source folder, then you will have issues with the dist path.

Reply:
maybe worth adding another API something like 
`getBrowserDistPathBySrcPath` which tomorrow will also work to distinguish between what we are going to put in the `main` prop of the package.json and the `browser` key in the package.json

Reply:
for `getDistPathBySrcPath` I just return the original path without the dist